Katello/katello
# Fixes #35006 - Remove runcible and conclude Pulp 2 removal

#### What are the changes introduced in this pull request?
Rips out all of the remaining Pulp 2 related code.

Some remains, like code relating to Docker push. We need to reimplement it with Pulp 3, so I figured that having the old code would serve as a good starting point. The APIs for users to interact with the code have been commented out, however.

#### Considerations taken when implementing this change?
A lot of things regarding sync status were removed, and it appears that none of them were in use. The sync status page still works. However, I'd like some help double checking that that all still works. Sync plans for example, would be good to test.

Similarly, the container registry had some changes. Looks like all is set, but some more eyes on it would be good. Perhaps I could further simplify some of it as well.

#### What are the testing steps for this pull request?
1) Make sure syncing all still works. This includes sync plans.
2) Make sure that the container registry still works.
3) Some package group related code was removed. Double check that there aren't floating APIs mentioning package group deletion or creation.
